test/e2e/utils/bundles.go

```diff
@@ -0,0 +1,164 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"github.com/containers/buildah"
+	"github.com/containers/buildah/pkg/blobcache"
+	ocilayout "github.com/containers/image/v5/oci/layout"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage"
+	"github.com/containers/storage/pkg/unshare"
+	"os"
+	"os/exec"
+	"path"
+	"sort"
+	"strings"
+	"time"
+)
+
+type Bundle struct {
+	PackageName             string
+	Version                 string
+	DefaultChannel          string
+	BundlePath              string
+	BundleManifestDirectory string
+	Channels                []string
+	GenerateAnnotations     bool
+}
+
+// Build the bundle image with opm and docker binary
+func (r *Registry) buildBundleImage(b *Bundle) (bundleReference string, err error){
+	if len(b.DefaultChannel) == 0 && len(b.Channels) == 0 {
+		return "", fmt.Errorf("missing default channel and channel list for package %s", b.PackageName)
+	}
+	if len(b.Channels) == 0 {
+		b.Channels = append(b.Channels, b.DefaultChannel)
+	}
+	if len(b.DefaultChannel) == 0 {
+		sort.Strings(b.Channels)
+		b.DefaultChannel = b.Channels[0]
+	}
+	bundlePath := b.BundlePath
+	if len(bundlePath) == 0 {
+		bundlePath = fmt.Sprintf("%s/operator", b.PackageName)
+	}
+	bundleReference = fmt.Sprintf("%s/%s:%s", r.URL, bundlePath, b.Version)
+
+	cmd := exec.Command("opm", "alpha", "bundle", "build", "-d", b.BundleManifestDirectory, "-b", "docker", "-e", b.DefaultChannel, "-c", strings.Join(b.Channels, ","), "--tag", bundleReference)
```

Comment:
I don't think docker is available as a container-tool within the e2e test-execution context. 

The container runtime on openshift is CRI-O -- I think we could maybe include podman in the e2e image and use that as the container-tool if we go down this route

Did we consider trying daemonless builds via containerd (ie using `none` as the container-tool)?

Reply:
The reason for using docker here was just to allow it to run locally, opm wasn't passing along the skipTLS option while pulling index images. Docker was the only container-tool enabling insecure pull by default. We can look at where the test is running before choosing a pull tool.